internal/place/international_address.go

```diff
@@ -0,0 +1,50 @@
+package place
+
+import "strings"
+
+type InternationalAddress struct {
+	ApartmentNumber string `json:"apartmentNumber"`
+	BuildingNumber  string `json:"buildingNumber"`
+	BuildingName    string `json:"buildingName"`
+	StreetName      string `json:"streetName"`
+	Town            string `json:"town"`
+	Region          string `json:"region"`
+	PostalCode      string `json:"postalCode"`
+	Country         string `json:"country"`
+}
+
+func (a InternationalAddress) ToAddress() Address {
+	var line1, line2, line3 string
+	if a.BuildingName != "" {
+		if a.ApartmentNumber != "" {
+			line1 = strings.TrimSpace(a.ApartmentNumber + ", " + a.BuildingName)
+		} else {
+			line1 = a.BuildingName
+		}
+
+		if a.BuildingNumber != "" {
+			line2 = strings.TrimSpace(a.BuildingNumber + " " + a.StreetName)
+		} else {
+			line2 = a.StreetName
+		}
+
+		line3 = a.Town
+	} else {
+		line1 = strings.TrimSpace(a.BuildingNumber + " " + a.StreetName)
+		if a.ApartmentNumber != "" {
+			line1 = a.ApartmentNumber + ", " + line1
+		}
+		line2 = a.Town
+	}
+
+	return Address{
+		Line1: line1,
+		Line2: line2,
+		Line3: line3,
+		// putting region here is weird, but don't really have a better place to
+		// store in the current structure
+		TownOrCity: a.Region,
```

Comment:
Tricky one...as a gut feeling having town and region together feels marginally better than region displayed as town. Obvs this is edge of edge case and may not matter so lets see how this tests with some real addresses.

Reply:
yeah we really need some real uses, I discussed with Greg and the fields aren't used for anything other than ordering the address. So (very) long term, maybe we'll move to a more generic "list of lines" format